execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -394,6 +479,238 @@ where
+                                let new_extension: Trie<K, V> = Trie::Extension {
+                                    affix: vec![sibling_idx].into(),
+                                    pointer: sibling_pointer,
+                                };
+                                let trie_key = Blake2bHash::new(&new_extension.to_bytes()?);
+                                new_elements.push((trie_key, new_extension))
+                            }
+                            // The single sibling is a extension.  We output an extension to replace
+                            // the parent, prepending the sibling index to the sibling's affix.  In
+                            // the next loop iteration, we will handle
+                            // the case where this extension might need to
+                            // be combined with a grandparent extension.
+                            Trie::Extension {
+                                affix: extension_affix,
+                                pointer,
+                            } => {
+                                let mut new_affix = vec![sibling_idx];
+                                new_affix.extend(Vec::<u8>::from(extension_affix));
+                                let new_extension: Trie<K, V> = Trie::Extension {
+                                    affix: new_affix.into(),
+                                    pointer,
+                                };
+                                let trie_key = Blake2bHash::new(&new_extension.to_bytes()?);
+                                new_elements.push((trie_key, new_extension))
+                            }
+                        }
+                    }
+                }
+            }
+            // The parent is a pointer block, and we are propagating a node or extension upwards. It
+            // is impossible to propagate a leaf upwards.  Reseat the thing we are propagating into
+            // the parent.
+            (Some((trie_key, _)), Trie::Node { mut pointer_block }) => {
+                let trie_node: Trie<K, V> = {
+                    pointer_block[idx as usize] = Some(Pointer::NodePointer(*trie_key));
+                    Trie::Node { pointer_block }
+                };
+                let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                new_elements.push((trie_key, trie_node))
+            }
+            // The parent is an extension, and we are outputting an extension.  Prepend the
+            // parent affix to affix of the output extension, mutating the output in place.
+            // This is the only mutate-in-place.
+            (
+                Some((
+                    trie_key,
+                    Trie::Extension {
+                        affix: child_affix,
+                        pointer,
+                    },
+                )),
+                Trie::Extension { affix, .. },
+            ) => {
+                let mut new_affix: Vec<u8> = Vec::<u8>::from(affix);
```

Comment:
No need to specify the type `Vec<u8>` twice.
```suggestion
                let mut new_affix: Vec<u8> = affix.into();
```
or
```suggestion
                let mut new_affix = Vec::<u8>::from(affix);
```

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/1155/commits/af8b66e01a5f65fe55260b1f7ed80742615528a8